src/H5FDsubfiling/H5subfiling_common.c

```diff
@@ -648,21 +658,40 @@ H5_open_subfiles(const char *base_filename, uint64_t file_id, H5FD_subfiling_par
     if (!context_id_out)
         H5_SUBFILING_GOTO_ERROR(H5E_FILE, H5E_CANTOPENFILE, FAIL, "invalid subfiling context ID pointer");
 
-    /* Initialize new subfiling context ID based on configuration information */
-    if (init_subfiling(base_filename, file_id, subfiling_config, file_acc_flags, file_comm, &context_id) < 0)
-        H5_SUBFILING_GOTO_ERROR(H5E_VFL, H5E_CANTINIT, FAIL, "couldn't initialize subfiling context");
+    /* Check if this file is already open */
+    H5E_BEGIN_TRY
+    {
+        status = H5_subfile_fid_to_context(file_id, &context_id);
```

Comment:
If it's the very first file open with the VFD, the opened file map won't be allocated yet and cause an error. Not a great design but the allocation of the map resides in the code that writes to it, which made decent sense at the time. Probably worth a slight refactoring.

Reply:
> If it's the very first file open with the VFD, the opened file map won't be allocated yet and cause an error. Not a great design but the allocation of the map resides in the code that writes to it, which made decent sense at the time. Probably worth a slight refactoring.

Yeah, I see that now.   Could you fix it by extracting this block from record_fid_to_subfile() into a separate helper routine, called by that routine and H5_subfile_fid_to_context():

```
   if (!sf_open_file_map) { 
        if (NULL == (sf_open_file_map = malloc((size_t)DEFAULT_FILE_MAP_ENTRIES *
            H5_SUBFILING_GOTO_ERROR(H5E_RESOURCE, H5E_CANTALLOC, FAIL, "couldn't

        sf_file_map_size = DEFAULT_FILE_MAP_ENTRIES;
        for (int i = 0; i < sf_file_map_size; i++) {
            sf_open_file_map[i].file_id       = UINT64_MAX;
            sf_open_file_map[i].sf_context_id = -1;
        }
    }
```

Because, if it's not an error, then H5_subfile_fid_to_context() should not return an error.  :-)